docs/eloquent-models/schema-builder.txt

```diff
@@ -0,0 +1,393 @@
+following changes to perform the schema changes on your MongoDB database:
+
+- Replace the ``Illuminate\Database\Schema\Blueprint`` import with
+  ``MongoDB\Laravel\Schema\Blueprint`` if it is referenced in your migration
+- Use only commands and syntax supported by {+odm-short+}
+
+.. tip::
+
+   If your default database connection is set to anything other than your
+   MongoDB database, update the following items so that PHP artisan performs
+   the migration on the correct database:
+
+   - Specify ``mongodb`` in the ``$connection`` field of your migration class
+   - Set ``DB_CONNECTION=mongodb`` in your ``.env`` configuration file
+
+The following example migration class contains the following methods:
+
+- ``up()``, which creates a collection and an index when you run the migration
+- ``down()``, which drops the collection and all the indexes on it when you roll back the migration
+
+.. literalinclude:: /includes/schema-builder/astronauts_migration.php
+   :dedent:
+   :language: php
+   :emphasize-lines: 6, 11
+
+Run or Roll Back a Migration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To run the database migration from a class file, run the following PHP artisan
+command after replacing the placeholder:
+
+.. code-block:: bash
+
+   php artisan migrate --path=<path to your migration class file>
+
+This artisan command runs the ``up()`` function in the class file to create
+the collection and index in the database specified in the ``config/database.php``
+file.
+
+To roll back the migration, run the following PHP artisan command after
+replacing the placeholder:
+
+.. code-block:: bash
+
+   php artisan migrate:rollback --path=<path to your migration class file>
+
+This command runs the ``down()`` function in the class file to drop the
+collection and related indexes.
+
+To learn more about Laravel migrations, see
+`Database: Migrations <https://laravel.com/docs/{+laravel-docs-version+}/migrations>`__
+in the Laravel documentation.
+
+.. _laravel-eloquent-collection-exists:
+
+Check Whether a Collection Exists
+---------------------------------
+
+To check whether a collection exists, call the ``hasCollection()`` method on
+the ``Schema`` facade in your migration file. You can use this to
+perform migration logic conditionally.
+
+The following example migration creates a ``stars`` collection if a collection
+named ``telescopes`` exists:
+
+.. literalinclude:: /includes/schema-builder/stars_migration.php
+   :language: php
+   :dedent:
+   :start-after: begin conditional create
+   :end-before: end conditional create
+
+.. _laravel-eloquent-indexes:
+
+Manage Indexes
+--------------
+
+MongoDB indexes are data structures that improve query efficiency by reducing
+the number of documents needed to retrieve query results. Certain indexes, such
+as geospatial indexes, extend how you can query the data.
+
+To improve query performance by using an index, make sure the index covers
+the query.  To learn more about indexes and query optimization, see the
```

Comment:
```suggestion
the query. To learn more about indexes and query optimization, see the
```

I assume style guide calls for a single space for sentences.

Reply:
Good catch! Will fix. 
Since Snooty ignores repeated whitespace (unless used for indentation of RST elements), there's no choice/rule.